### PR TITLE
Add support for Amazon Linux 2

### PR DESCRIPTION
**Pull request by pschaumburg**

### CODE REVIEW

**Comment by pschaumburg** [comment]
Finally fixed some lints and added spec tests.

**Comment by pschaumburg** [comment]
@tas50

File: spec/kitchen/driver/ec2/image_selection_spec.rb [hunk 1]
```diff
@@ -68,6 +68,19 @@ def new_instance(platform_name: "blarghle")
       { :name => "architecture", :values => ["x86_64"] },
     ],
 
+    "amazon2" => [
+      { :name => "owner-id", :values => %w{137112412989} },
+      { :name => "name", :values => %w{amzn2-ami-hvm-2.0.*} },
+    ],
+    "amazon2-2017" => [
```
Comment (tas50) [inline spec/kitchen/driver/ec2/image_selection_spec.rb]: Do we really need to support the 2017 amazon linux. That was just the RC release where they used the old naming schema. They haven't made it entirely clear how they plan to name the new 2.x series.

File: README.md [hunk 2]
```diff
@@ -61,6 +61,8 @@ platforms:
   - name: centos-6.3
   # The latest patch release of Amazon Linux 2017.03
   - name: amazon-2017.03
+  # The latest patch release of Amazon Linux 2 2018
+  - name: amazon-2018
```
Comment (tas50) [inline README.md]: This seems like the wrong UX for selecting amazon 2

**Comment by pschaumburg** [comment]
@tas50 Changed the necessary files to allow only amazon linux 2 stable releases with the official namings from aws.

**Comment by tas50** [comment]
@pschaumburg Rebase and this should be good

**Comment by pschaumburg** [comment]
@tas50 Here we go ;)

**Comment by tas50** [comment]
Thanks

**Pull request merged**